### PR TITLE
Add webapp and webapi in one app integration testcase

### PR DESCRIPTION
**Pull request by moarychan**
Fixes Azure/azure-sdk-for-java#19816

Feature PR: https://github.com/Azure/azure-sdk-for-java/pull/22674

Changelog:
- Add ITs for configured application type `web_application_and_resource_server`.
- Sort lines for the application type.
- Rename artfactId `azure-spring-boot-test-aad-obo` to `azure-spring-boot-test-aad-webapp-and-resource-server-with-obo`.

### CODE REVIEW

File: sdk/spring/azure-spring-boot-test-aad-obo/pom.xml [hunk 1]
```diff
@@ -49,6 +49,12 @@
       <artifactId>spring-boot-starter-test</artifactId>
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-test-aad-obo/pom.xml:49]: To make less confusing, how about rename this module to `azure-spring-boot-test-aad-webapp-and-resource-server-with-obo`?

File: sdk/spring/azure-spring-boot-test-aad/src/test/java/com/azure/test/aad/selenium/AADSeleniumITHelper.java [hunk 2]
```diff
@@ -39,74 +32,12 @@
     }
 
     public AADSeleniumITHelper(Class<?> appClass, Map<String, String> properties) {
-        this(appClass, properties, AAD_USER_NAME_1, AAD_USER_PASSWORD_1);
+        super(appClass, properties, AZURE_CLOUD_TYPE,
+            AAD_TENANT_ID_1, AAD_USER_NAME_1, AAD_USER_PASSWORD_1);
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-test-aad/src/test/java/com/azure/test/aad/selenium/AADSeleniumITHelper.java:36]: nit: `AADSeleniumITHelper(appClass, properties, AZURE_CLOUD_TYPE,
            AAD_TENANT_ID_1, AAD_USER_NAME_1, AAD_USER_PASSWORD_1)`

File: sdk/spring/azure-spring-boot-test-aad-obo/src/test/java/com/azure/test/aad/selenium/AADWebAppAndWebApiInOneAppIT.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+public class AADWebAppAndWebApiInOneAppIT {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AADWebAppAndWebApiInOneAppIT.class);
+    private static final String GRAPH_ME_ENDPOINT = "https://graph.microsoft.com/v1.0/me";
+
+    private Map<String, String> properties;
+
+    @BeforeAll
+    public void beforeAll() {
+        properties = createDefaultProperties();
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-test-aad-obo/src/test/java/com/azure/test/aad/selenium/AADWebAppAndWebApiInOneAppIT.java:58]: To make things simple, don't create abstract class. Just create `createProperties()` in `AADWebAppAndWebApiInOneAppIT`.

File: sdk/spring/azure-spring-boot-test-aad-obo/src/test/java/com/azure/test/aad/selenium/AADSeleniumITHelper.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import java.util.Map;
+
+import static com.azure.spring.test.EnvironmentVariable.AAD_MULTI_TENANT_CLIENT_ID;
+import static com.azure.spring.test.EnvironmentVariable.AAD_MULTI_TENANT_CLIENT_SECRET;
+import static com.azure.spring.test.EnvironmentVariable.AAD_TENANT_ID_1;
+import static com.azure.spring.test.EnvironmentVariable.AZURE_CLOUD_TYPE;
+
+public class AADSeleniumITHelper extends AbstractAADSeleniumITHelper {
+
+    public static Map<String, String> createDefaultProperties() {
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-test-aad-obo/src/test/java/com/azure/test/aad/selenium/AADSeleniumITHelper.java:18]: If the only difference is this method, we can put this method in test.

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/spring/azure-spring-boot-test-aad-obo/src/test/java/com/azure/test/aad/selenium/AADWebAppAndWebApiInOneAppIT.java (hunk 3 above)
Comment (moarychan) [reply sdk/spring/azure-spring-boot-test-aad-obo/src/test/java/com/azure/test/aad/selenium/AADWebAppAndWebApiInOneAppIT.java:58]: OK

File: sdk/spring/azure-spring-boot-test-selenium-common/src/main/java/com/azure/test/aad/common/AADSeleniumITHelper.java [hunk 5]
```diff
@@ -1,18 +1,16 @@
-package com.azure.test.aad.selenium;
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.test.aad.common;
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-test-selenium-common/src/main/java/com/azure/test/aad/common/AADSeleniumITHelper.java:3]: Why  change package name?

File: sdk/spring/azure-spring-boot-test-aad/src/test/java/com/azure/test/aad/selenium/AADITHelper.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.Map;
+
+import static com.azure.spring.test.EnvironmentVariable.AAD_SINGLE_TENANT_CLIENT_ID;
+import static com.azure.spring.test.EnvironmentVariable.AAD_SINGLE_TENANT_CLIENT_SECRET;
+import static com.azure.spring.test.EnvironmentVariable.AAD_TENANT_ID_1;
+import static com.azure.spring.test.EnvironmentVariable.AZURE_CLOUD_TYPE;
+
+public class AADITHelper {
+
+    public static Map<String, String> createDefaultProperties() {
```
Comment (chenrujun) [inline sdk/spring/azure-spring-boot-test-aad/src/test/java/com/azure/test/aad/selenium/AADITHelper.java:17]: Why extract this to a new class?

File: sdk/spring/azure-spring-boot-test-selenium-common/src/main/java/com/azure/test/aad/common/AADSeleniumITHelper.java (hunk 5 above)
Comment (moarychan) [reply sdk/spring/azure-spring-boot-test-selenium-common/src/main/java/com/azure/test/aad/common/AADSeleniumITHelper.java:3]: This class is the common helper for AAD selenium in the selenium-common project, it's in the same location with `SeleniumITHelper`, you mean it should keep the package name 'com.azure.test.aad.selenium.AADSeleniumITHelper', or change to `com.azure.test.aad.common.selenium.AADSeleniumITHelper`?

File: sdk/spring/azure-spring-boot-test-aad/src/test/java/com/azure/test/aad/selenium/AADITHelper.java (hunk 6 above)
Comment (moarychan) [reply sdk/spring/azure-spring-boot-test-aad/src/test/java/com/azure/test/aad/selenium/AADITHelper.java:17]: There are many references, so I extracted them here.

Comment (chenrujun) [reply sdk/spring/azure-spring-boot-test-aad/src/test/java/com/azure/test/aad/selenium/AADITHelper.java:17]: Can we keep original `AADSeleniumITHelper`?

**Review by chenrujun — APPROVED** [review]
LGTM.

**Pull request merged**